salt/modules/aptpkg.py

```diff
@@ -1686,7 +1686,10 @@ def _get_opts(line):
     Return all opts in [] for a repo line
     """
     get_opts = re.search(r"\[.*\]", line)
-    ret = {"arch": {"full": "", "value": ""}, "signedby": {"full": "", "value": ""}}
+    ret = {
+        "arch": {"full": "", "value": "", "index": 0},
+        "signedby": {"full": "", "value": "", "index": 0},
```

Comment:
As mentioned in #62176, this entry is only using `signed-by`:

```console
deb [signed-by=/usr/share/postgresql-common/pgdg/apt.postgresql.org.gpg] http://apt.postgresql.org/pub/repos/apt buster-pgdg main
```

Which hits a traceback on the last line here:

https://github.com/saltstack/salt/blob/8253b6cef00aaa97db0dac1d46b212c1bf220df9/salt/modules/aptpkg.py#L2999-L3002

In `_get_opts`, there's no guarantee that an `index` will be set for each by this block (e.g. when only the `arch` or the `signed-by` option is provided in the entry, as shown above):

https://github.com/saltstack/salt/blob/8253b6cef00aaa97db0dac1d46b212c1bf220df9/salt/modules/aptpkg.py#L1696-L1704

So a default `index` is provided for each.